source/constructs/api/data_source/service.py

```diff
@@ -1589,6 +1602,10 @@ def query_glue_connections(account: AdminAccountInfo):
                                                                            MaxResults=100,
                                                                            HidePassword=True)['ConnectionList']
 
+def query_glue_databases(account: AdminAccountInfo):
+    glue_client = boto3.client('glue', region_name=_admin_account_region)
+    return glue_client.get_databases()['DatabaseList']
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The API method `get_databases` returns paginated results instead of all results. Consider using the pagination API or checking one of the following keys in the response to verify that all results were returned: `IsTruncated`, `NextToken`.

[Learn more](https://boto3.amazonaws.com/v1/documentation/api/latest/guide/paginators.html)